src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -0,0 +1,366 @@
+            }
+            else if (token.Phase < 3)
+            {
+                newContinuationToken = EverythingOperationContinuationToken.ToString(token.Phase + 1, null);
+            }
+
+            return new SearchResult(searchResult.Results, newContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
+        }
+
+        private async Task<SearchResult> SearchIncludes(
+            string resourceId,
+            PartialDateTime since,
+            IReadOnlyList<string> types,
+            CancellationToken cancellationToken)
+        {
+            using IScoped<ISearchService> search = _searchServiceFactory();
+            var searchResultEntries = new List<SearchResultEntry>();
+
+            // Build search parameters
+            var searchParameters = new List<Tuple<string, string>>
+            {
+                Tuple.Create(SearchParameterNames.Id, resourceId),
+            };
+
+            searchParameters.AddRange(_includes.Select(include => Tuple.Create(SearchParameterNames.Include, $"{ResourceType.Patient}:{include}")));
+
+            // Search
+            SearchOptions searchOptions = _searchOptionsFactory.Create(ResourceType.Patient.ToString(), searchParameters);
+            SearchResult searchResult = await search.Value.SearchAsync(searchOptions, cancellationToken);
+            searchResultEntries.AddRange(searchResult.Results.Select(x => new SearchResultEntry(x.Resource)));
+
+            // Filter results by IsDeleted
+            searchResultEntries = searchResultEntries.Where(s => !s.Resource.IsDeleted).ToList();
+
+            // Filter results by _type
+            if (types.Any())
+            {
+                searchResultEntries = searchResultEntries.Where(s => types.Contains(s.Resource.ResourceTypeName)).ToList();
+            }
+
+            // Filter results by _since
+            if (since != null)
+            {
+                var sinceDateTimeOffset = since.ToDateTimeOffset(
+                    defaultMonth: 1,
+                    defaultDaySelector: (year, month) => 1,
+                    defaultHour: 0,
+                    defaultMinute: 0,
+                    defaultSecond: 0,
+                    defaultFraction: 0.0000000m,
+                    defaultUtcOffset: TimeSpan.Zero);
+                searchResultEntries = searchResultEntries.Where(s => s.Resource.LastModified.CompareTo(sinceDateTimeOffset) >= 0).ToList();
+            }
+
+            return new SearchResult(searchResultEntries, searchResult.ContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
+        }
+
+        private async Task<SearchResult> SearchRevinclude(
+            string resourceId,
+            PartialDateTime since,
+            IReadOnlyList<string> types,
+            string continuationToken,
+            CancellationToken cancellationToken)
+        {
+            if (types.Any() && !types.Contains(_revinclude.Item1))
```

Comment:
>_revinclude.Item1 [](http://example.com/codeflow?start=47&length=17)

nit: This is so hard to read.
I would break that tuple into two const strings and use them.

`!types.Contains(Device)`
or 
`Tuple.Create(Patient, resourceId)`
is more readable than

`!types.Contains(_revinclude.Item1)`
and
`Tuple.Create(_revinclude.Item2, resourceId)`

Reply:
Or if you plan to have collection of revInclude can you use
`private readonly (string ResourceType, string SearchCode) _revinclude = new ("Device", "patient");
instead?

Reply:
Agree! It is fixed in the update.